src/main/java/org/opensearch/flowframework/transport/ProvisionWorkflowTransportAction.java

```diff
@@ -266,7 +266,7 @@ private void executeWorkflow(List<ProcessNode> workflowSequence, String workflow
                 status = ExceptionsHelper.status(ex);
             }
             logger.error("Provisioning failed for workflow {} during step {}.", workflowId, currentStepId, ex);
-            String errorMessage = (ex.getCause() == null ? ex.getClass().getName() : ex.getCause().getClass().getName())
+            String errorMessage = (ex.getCause() == null ? ex.getMessage() : ex.getCause().getClass().getName())
```

Comment:
I don't see how we're testing the "safe" exception here?

Reply:
For any step we are checking if the exception is safe if not then have a custom message for that step. Either way both the messages should be safe to expose.

```java
                   Exception e = getSafeException(ex);
                    String errorMessage = (e == null ? "Failed to register remote model" : e.getMessage());
                    logger.error(errorMessage, e);
                    registerRemoteModelFuture.onFailure(new WorkflowStepException(errorMessage, ExceptionsHelper.status(e)));
```

Reply:
Would it be safer to have safe exception check again here in case a step was missed?